### PR TITLE
Update/Shareable Links Approval Updates

### PR DESCRIPTION
**Pull request by altaudio**
### 💬 Description
Adds changes to a selection of gather-ui components.

This seems like a massive PR, but it is actually multiple individual changes to several different components.

Also adds `<ApprovedLabel />` component.

#### Refactors
- StatusIndicator story refactored and moved next to component file.
- Corrects `pluralisePerson` typo
### 🔗 Links
Changes relate to the following designs:
https://www.figma.com/file/QIUD7WxupVObMVYCG2hFJ3uQ/Shareable-URLs-%E2%80%93-Master?node-id=369%3A3482

https://www.figma.com/file/QIUD7WxupVObMVYCG2hFJ3uQ/Shareable-URLs-%E2%80%93-Master?node-id=375%3A0
### 🚪 Start Points
No top down start point, just changes to individual components
### ✅ Checklist
- [ ] Tests written
- [ ] Browser tested
- [x] Added to documentation
- [x] Added to storybook

### CODE REVIEW

**Comment by altaudio** [comment]
[Approvals: Item Editor UI](https://trello.com/c/a8UTXHKn/596-approvals-item-editor-ui)

File: lib/Icon/index.js [hunk 1]
```diff
@@ -168,34 +172,47 @@ const getIcon = name => {
...
     question: questionSVG,
     hierarchy: hierarchySVG,
     userInvite: userInviteSVG
   };
   return icons[name] ? icons[name] : () => null;
 };
 
+const createClassesFromTypesList = (types, prefix = 'icon--') =>
+  types.reduce((className, type) => `${className} ${prefix}${type}`, '');
+
```
Comment (AmeeMorris) [inline lib/Icon/index.js]: Goes without saying, but make sure you catch all uses of this on the app!

File: lib/MentionsForm/MentionsFormActions.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import cx from 'classnames';
 import PropTypes from 'prop-types';
 import Button from '../Button';
-import { pluarisePerson } from '../helpers';
+import { pluralisePerson } from '../helpers';
```
Comment (AmeeMorris) [inline lib/MentionsForm/MentionsFormActions.js]: 😂

File: lib/StatusIndicator/index.js [hunk 3]
```diff
@@ -36,6 +38,10 @@ const StatusIndicator = ({
     'status-indicator--row': row
   });
 
+  const circleClassNames = cx('status-indicator__circle', {
+    'status-indicator__circle--large': circleLarge || approved
+  });
+
```
Comment (AmeeMorris) [inline lib/StatusIndicator/index.js]: Yeah this one I dont have an issue with, cos it'd be a right pain to override!

File: lib/ExpandingTextArea/styles.scss [hunk 4]
```diff
@@ -6,8 +6,13 @@
...
   transition: border .1s ease;
+
   &:focus {
     outline: none;
     border: 1px solid $color-border-base-hover;
   }
+
+  &--width-medium {
+    width: $layout-spacing-base * 10;
+  }
```
[the remark this answers is not in the document]
Comment (altaudio) [reply lib/ExpandingTextArea/styles.scss]: Totally agree, this is too specific. I'll look to make a change to this 👍

File: lib/Icon/index.js (hunk 1 above)
Comment (altaudio) [reply lib/Icon/index.js]: Actually I forgot to do this, so probably doesn't go without saying! 😆

**Comment by altaudio** [comment]
@AmeeMorris Thanks for the comments! I'v added both of those changes as they were good points.

Could you check the new SASS helper class is ok? I haven't played with them before.

**Pull request merged**